frontend/hooks/data-fetching/course.ts

```diff
@@ -330,10 +330,9 @@ export async function createQueue(
 
     if (!res.ok) {
         const error = await res.json();
-        throw {
-            payload: JSON.stringify(payload),
-            error: JSON.stringify(error),
-        };
+        const errorObj = Error(JSON.stringify(error));
+        logException(errorObj, JSON.stringify(payload));
```

Comment:
Not sure if this is planned in the future, but I'm not sure if it's correct to have inconsistent behavior between `createQueue` and `createQuestion`. In particular, `createQuestion` will no longer throw an exception while `createQueue` and `finishQuestion` will, and the return types are also different. Would it make sense for all these functions to return `Promise<[number, any]>`, where the number is the status code and any is the error object or undefined if it is a success? Could be even better if we define some type that is a discriminated union where a success status code does not require an error object. 

This does sound like a bigger refactor, and I don't mind letting this through for now.

Reply:
Theoretically we should do this but we decided to just refactor the code that is throwing errors for now, so I think this is getting put in the backlog for now.